### DISCUSSION TITLE
Auto-Reload on a timer.

### DISCUSSION THREAD

**Discussion by Fezalion** [Beastwick18/nyaa#19] [Ideas]
Hi, I am currently using nyaa on Windows with windows terminal on a separate monitor to see latest torrents on nyaa.si, is there a way for me to reload for example every 5 minutes automaticaly, either within the app itself or within Windows Terminal, any help is appreciated.

**Comment by Beastwick18** [comment]
Currently, there is no built-in way to do this. However, it should be possible with something like tmux, assuming you're using WSL.

You could first open tmux, which will launch a new session with a green bar at the bottom. You can then open nyaa from within tmux. Then, open a new Windows Terminal tab thats using the same WSL Linux Distro and run the command:
```
tmux send-keys r
```
to make sure it works. This should trigger the <kbd>r</kbd> key on the tmux session.

Assuming that works, you could make the desired feature into a bash script (a `.sh` file) that refreshes every X seconds. Something like:
```sh
#!/bin/sh

while [ true ]; do
	tmux send-keys r
	sleep 5m # Wait for 5 minutes before pressing 'r' again
done
```

Then, while nyaa is running in the tmux tab, run the script from the second tab with bash, like:
```sh
bash test.sh # or whatever you've named the script
```

To close out of the script, <kbd>Ctrl</kbd>+<kbd>C</kbd> should work.

**Reply by Beastwick18** [reply]
Something like
```sh
wt.exe --title "Auto refresh" -- wsl.exe -d Arch "/usr/bin/bash" "/path/to/test.sh" && tmux new-session 'nyaa'
```
should do everything at once: spawn a new tab in Windows Terminal with the refresh script, and start a new tmux session with nyaa already running.

Replace `Arch` with your WSL Distro, e.g. Ubuntu. And replace `/path/to/test.sh` with the path to your refresh script.

**Comment by Fezalion** [comment]
Thanks, that works for me.
